### PR TITLE
Inji 235 get issuer config endpoint response modification

### PR DESCRIPTION
**Pull request by KiruthikaJeyashankar**
Return "bad request" response when invalid issuer id is passed to "get issuer config" (/issuers/{invalid-IssuerId}) endpoint

### CODE REVIEW

File: src/main/java/io/mosip/mimoto/exception/InvalidIssuerIdException.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package io.mosip.mimoto.exception;
+
+public class InvalidIssuerIdException extends BaseUncheckedException {
+
+    /**
+     * Unique id for serialization
+     */
+    private static final long serialVersionUID = -5350213197226295789L;
```
Comment (KiruthikaJeyashankar) [inline src/main/java/io/mosip/mimoto/exception/InvalidIssuerIdException.java:8]: serialVersionUID's use case yet to be figured out.
(This UID in code is copied from InvalidInputException)

File: src/main/java/io/mosip/mimoto/controller/IssuersController.java [hunk 2]
```diff
@@ -33,13 +39,17 @@ public ResponseEntity<Object> getAllIssuers() {
...
-        //TODO: Modify id
-        responseWrapper.setId("mosip.inji.properties");
+        responseWrapper.setId(ID);
         responseWrapper.setVersion("v1");
         responseWrapper.setResponsetime(DateUtils.getRequestTimeString());
 
 
-        responseWrapper.setResponse(issuersService.getIssuerConfig(issuerId));
+        IssuerDTO issuerConfig = issuersService.getIssuerConfig(issuerId);
+        responseWrapper.setResponse(issuerConfig);
```
Comment (swatigoel) [inline src/main/java/io/mosip/mimoto/controller/IssuersController.java:48]: Why are we setting response here?
If it's an error, then there should not be response field available

File: src/test/java/io/mosip/mimoto/service/IssuersServiceTest.java [hunk 3]
```diff
@@ -36,7 +36,7 @@ public class IssuersServiceTest {
     List<String> issuerConfigRelatedFields = List.of("additionalHeaders", "serviceConfiguration", "redirectionUri");
 
 
-    private static IssuerDTO getIssuerDTO(String issuerName, List<String> nullFields) {
+    public static IssuerDTO getIssuerDTO(String issuerName, List<String> nullFields) {
```
Comment (swatigoel) [inline src/test/java/io/mosip/mimoto/service/IssuersServiceTest.java:39]: Do not use service DTO in controller.

File: src/test/java/io/mosip/mimoto/controller/InjiControllerTest.java [hunk 4]
```diff
@@ -36,9 +40,9 @@
 
 import java.io.IOException;
 import java.nio.file.Files;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
 
+import static io.mosip.mimoto.service.IssuersServiceTest.getIssuerDTO;
```
Comment (swatigoel) [inline src/test/java/io/mosip/mimoto/controller/InjiControllerTest.java:45]: Why are we referring to IssuersServiceTest for DTO?
create DTO required for controller.

**Review by swatigoel — CHANGES_REQUESTED** [review]

File: src/main/java/io/mosip/mimoto/controller/IssuersController.java [hunk 5]
```diff
@@ -33,13 +39,17 @@ public ResponseEntity<Object> getAllIssuers() {
...
+        if (issuerConfig == null) {
+            responseWrapper.setErrors(List.of(new ErrorDTO(PlatformErrorMessages.INVALID_ISSUER_ID_EXCEPTION.getCode(), PlatformErrorMessages.INVALID_ISSUER_ID_EXCEPTION.getMessage())));
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(responseWrapper);
+        }
```
Comment (swatigoel) [inline src/main/java/io/mosip/mimoto/controller/IssuersController.java:52]: Let's use http status 404, as 404 for URL requested is invalid or the resource requested, such as an item or a user, does not exist

**Review by swatigoel — CHANGES_REQUESTED** [review]

File: src/main/java/io/mosip/mimoto/controller/IssuersController.java (hunk 2 above)
Comment (KiruthikaJeyashankar) [reply src/main/java/io/mosip/mimoto/controller/IssuersController.java:48]: Refactored

File: src/main/java/io/mosip/mimoto/controller/IssuersController.java (hunk 5 above)
Comment (KiruthikaJeyashankar) [reply src/main/java/io/mosip/mimoto/controller/IssuersController.java:51]: Modified

File: src/test/java/io/mosip/mimoto/controller/InjiControllerTest.java (hunk 4 above)
Comment (KiruthikaJeyashankar) [reply src/test/java/io/mosip/mimoto/controller/InjiControllerTest.java]: Refactored

File: src/test/java/io/mosip/mimoto/service/IssuersServiceTest.java (hunk 3 above)
Comment (KiruthikaJeyashankar) [reply src/test/java/io/mosip/mimoto/service/IssuersServiceTest.java]: Refactored

File: src/main/java/io/mosip/mimoto/exception/InvalidIssuerIdException.java (hunk 1 above)
Comment (KiruthikaJeyashankar) [reply src/main/java/io/mosip/mimoto/exception/InvalidIssuerIdException.java]: Updated serialVersionUID

File: src/main/java/io/mosip/mimoto/controller/IssuersController.java (hunk 2 above)
Comment (swatigoel) [reply src/main/java/io/mosip/mimoto/controller/IssuersController.java:48]: I was saying, error response should not have field for 2xx response.
so here error response should be like with other attributes like status:404 if needed
```
{
    "errorCode":"code",
    "errorMessage":"message"
}
```

**Review by swatigoel — APPROVED** [review]

**Pull request merged**